src/components/DetailedCharacter.jsx

```diff
@@ -0,0 +1,23 @@
+import styles from './DetailedCharacter.css';
+
+export default function DetailedCharacter({ characters, id }) {
+  const character = characters[id];
```

Comment:
this feels like a fragile way to display your detailed info - the preferred method in this case would be to fetch character details from the api in a `useEffect` in the parent component and then pass that character object in as props. for instance, if your api changed ids to hashes or the ids were no longer sequential then the id would not match up to the index of that character. forgive me if you had a reason for doing it this way that I'm just not seeing!